code/obj/item/sec_tape.dm

```diff
@@ -0,0 +1,181 @@
+	amount = STARTAMOUNT
+	max_stack = MAXTAPE
+	stack_type = /obj/item/sec_tape
+	icon = 'icons/obj/sec_tape.dmi'
+	icon_state = "sec_tape_roll"
+	w_class = W_CLASS_TINY
+	flags = TABLEPASS|EXTRADELAY|FPRINT|CONDUCT|ONBELT
+	rand_pos = 1
+	event_handler_flags = USE_GRAB_CHOKE | USE_FLUID_ENTER
+	special_grab = /obj/item/grab
+	inventory_counter_enabled = 1
+
+	New(loc, length = STARTAMOUNT)
+		src.amount = length
+		pixel_x = rand(-2,2)
+		pixel_y = rand(-2,2)
+		..(loc)
+		BLOCK_SETUP(BLOCK_ROPE)
+
+	before_stack(atom/movable/O as obj, mob/user as mob)
+		user.visible_message("<span class='notice'>[user] begins awkwardly sticking the tape back onto the roll.</span>")
+
+	after_stack(atom/movable/O as obj, mob/user as mob, var/added)
+		boutput(user, "<span class='notice'>You finish sticking the tape back together.</span>")
+
+	custom_suicide = 1
+	suicide(var/mob/user as mob)
+		if (!src.user_can_suicide(user))
+			return 0
+		//Are we on valid turf?
+		var/turf/T = get_turf(user)
+		if (isturf(user.loc) && istype(T, /turf/simulated/floor))
+			//Setup the perfect crime
+			user.visible_message("<span class='alert'><b>[user] carefully sets up some security tape around themselves then swallows the tape and chokes on it, satisfied that they created the perfect crime scene.</b></span>")
+			var/obj/sec_tape/T1 = new /obj/sec_tape(T, src)
+			T1.dir = EAST
+			T1.tape_UpdateIcon()
+			T1.layerify()
+			var/obj/sec_tape/T2 = new /obj/sec_tape(T, src)
+			T2.dir = WEST
+			T2.tape_UpdateIcon()
+			T2.layerify()
+			var/obj/sec_tape/T3 = new /obj/sec_tape(T, src)
+			T3.dir = NORTH
+			T3.tape_UpdateIcon()
+			T3.layerify()
+			var/obj/sec_tape/T4 = new /obj/sec_tape(T, src)
+			T4.tape_UpdateIcon()
+			T4.layerify()
+			//No need to set it up to the south, it already is by default
+			user.death(FALSE)
+			qdel(src)
+		else
+			//Else we'll choke ourselves out
+			user.visible_message("<span class='alert'><b>[user] wraps some tape around \his neck and tightens it.</b></span>")
+			user.take_oxygen_deprivation(160)
+			SPAWN(50 SECONDS)
+				if (user && !isdead(user))
+					user.suiciding = 0
+		return 1
+
+	proc/use(var/used) // remove "used" amount from the coil
+		if (src.amount < used)
+			return 0
+		else if (src.amount == used)
+			qdel(src)
+			return 1
+		else
+			amount -= used
+			tooltip_rebuild = 1
+			src.UpdateIcon()
+			return 1
+
+	proc/take(var/amt, var/newloc) // removes "amt" from the coil and put it somewhere, use for coil splitting with wirecutters
+		if (amt > amount)
+			amt = amount
+			tooltip_rebuild = 1
+			src.UpdateIcon()
+		if (amt == amount)
+			if (ismob(loc))
+				var/mob/owner = loc
+				owner.u_equip(src)
+			set_loc(newloc)
+			return src
+		src.use(amt)
+		var/obj/item/sec_tape/C = new /obj/item/sec_tape(newloc)
+		C.amount = amt
+		return
+
+	update_icon()
+		inventory_counter?.update_number(amount)
+		return
+
+/obj/item/sec_tape/vended //Tape in secvends has a pre-set amount of uses
+	New(loc, length)
```

Comment:
```suggestion
	New(loc)
```
feels weird to have an argument that does nothing

Reply:
removed it.